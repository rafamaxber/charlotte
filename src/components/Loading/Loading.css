.img-loading {
  width: 50px;
  margin: auto;
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  top: 30px;
  z-index: 10;
}