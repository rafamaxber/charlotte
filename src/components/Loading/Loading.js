/* TODO: 
*  [] Improve load svg icon by sprite svg
*  [] Create other loading components to especific load
*/
import React from 'react';
import styled from 'styled-components';
import './Loading.css';
import svgLogo from '../../assets/loading-icon.svg';

const loadingBar = styled.div`
  position: relative;
`;

const Loading = () => (
  <loadingBar>
    <img className="img-loading" src={svgLogo} alt="loading..."/>
  </loadingBar>
);

export default Loading;
