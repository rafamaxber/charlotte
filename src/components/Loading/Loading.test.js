import React from 'react';
import renderer from 'react-test-renderer';
import Loadable from './Loadable';
import Loading from './Loading';

describe('Loading components', () => {
  describe('Loadable', () => {
    it('Should match with snapshot', () => {
      expect(Loadable()).toMatchSnapshot();
    });
  });
  describe('Loadable', () => {
    it('Should match with snapshot', () => {
      const component = renderer.create(
        <Loading></Loading>
      );
      expect(component).toMatchSnapshot();
    });
  });
});
