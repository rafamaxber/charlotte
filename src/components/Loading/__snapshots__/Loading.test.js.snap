// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading components Loadable Should match with snapshot 1`] = `[Function]`;

exports[`Loading components Loadable Should match with snapshot 2`] = `
<loadingBar>
  <img
    alt="loading..."
    className="img-loading"
    src="loading-icon.svg"
  />
</loadingBar>
`;
